src/Stack/Prelude.hs

```diff
@@ -76,8 +76,7 @@ sinkProcessStderrStdout
   -> RIO env (e,o)
 sinkProcessStderrStdout name args sinkStderr sinkStdout =
   withProc name args $ \pc0 -> do
-    let pc = setStdin closed
```

Comment:
I'm not in favor of changing this default—which IMO is the right default—for this one case, which should be considered an exception. I preferred the previous approach. If the goal is to reduce code duplication, I would recommend changing the function to take a `ProcessConfig` instead, or something along those lines.

Reply:
Orthogonal to this PR... I think we _do_ have to change this default. I found another part of the code base which is expecting a working `stdin` in a subprocess: configuring a package with `build-type: Configure`. So let's move ahead with this as is.